src/SpeakDirectiveParser.test.js

```diff
@@ -0,0 +1,75 @@
+import { extractAlexaTextResponse } from "./SpeakDirectiveParser";
+import IllegalArgumentError from "./errors/IllegalArgumentError";
+
+import testData from "./test-data/multipart-response-test-data";
+
+it("throws an error if an empty string is passed as input", () => {
```

Comment:
Add tests to handle empty/undefined results from *httpMessageParser*. Also test cases where the key paths for parsed message are missing from the response.